packages/application/src/test/application.spec.ts

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2018-2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as nbformat from '@jupyterlab/nbformat';
+import { NotebookModel } from '@jupyterlab/notebook';
+import { JupyterServer, NBTestUtils } from '@jupyterlab/testutils';
+
+import { NotebookParser } from '../parsing';
+import { RequestHandler } from '../requests';
+import { FrontendServices } from '../services';
+
+const server = new JupyterServer();
+const defaultContent: any = {
+  cells: [
+    {
+      cell_type: 'code',
+      execution_count: null,
+      metadata: {},
+      outputs: [],
+      source: ['import os\n', "print(os.environ['HOME'])"]
+    }
+  ],
+  metadata: {
+    kernelspec: {
+      display_name: 'Python 3',
+      language: 'python',
+      name: 'python3'
+    },
+    language_info: {
+      codemirror_mode: {
+        name: 'ipython',
+        version: 3
+      },
+      file_extension: '.py',
+      mimetype: 'text/x-python',
+      name: 'python',
+      nbconvert_exporter: 'python',
+      pygments_lexer: 'ipython3',
+      version: '3.7.6'
+    }
+  },
+  nbformat: 4,
+  nbformat_minor: 4
+};
+const codeSnippetSchema = {
```

Comment:
is this the same schema we have on the server? would this need to be always in-sync? If yes, is there a way to read that from the server?